python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -209,3 +383,201 @@ def proc_to_examine():
     max_mem_bytes = 608_662_528
 
     check_process_memory_leaks(proc_to_examine, 20, max_mem_bytes, 80.0)
+
+#endregion
+
+#region TESTS pytest-memray type for memory limit and leaks
+
+## NOTE: Currently tests can be executed on Python >= 3.8 only
+
+
+symbol = "test"
+
+
+@pytest.fixture
+def library_with_symbol(arctic_library_lmdb) -> Generator[Tuple[Library, pd.DataFrame], None, None]:
+    """
+        As memray instruments memory, we need to take out 
+        everything not relevant from mem leak measurement out of 
+        test, so it works as less as possible
+    """
+    lib: Library = arctic_library_lmdb
+    df = construct_df_querybuilder_tests(size=2000000)
+    lib.write(symbol, df)
+    yield (lib, df)
+
+def mem_query(lib: Library, df: pd.DataFrame, num_repetitions:int=1, read_batch:bool=False):
```

Comment:
`pytest-memray` docs recommend running in a loop, should we set `num_repetitions>1` for fast tests?

Reply:
mem leaks will be revealed over long running repeated processes indeed. As our dafarame is >16million all queries are likely to trigger extreme number of repetitions in C++ that should effectivley achieve that.

Alternative is to have smaller dataframe 30-50k symbols and repeat queries many times. This test will capture other types of mem leaks perhaps. 

But without knowing the code my inclination would be that more serious would be the leaks that current test will capture.

But at the same time it is really cheap to add now additional tests with more repetitions. That will only cost couple of more minutes time if no problems arise

Reply:
I think both a single iteration with a large dataframe, and many iterations with smaller dataframes have value, as they could expose different styles of leak.
e.g. if we leak 1kb of memory every time we do a read with a filter, this will only be caught by running many filters

Reply:
Will add test

Reply:
And actually there seems to be a problem https://man312219.monday.com/boards/7852509418/pulses/8067881190
I will xfail and push the code